### PR TITLE
chore(explore): Visual updates to explore datasource panel

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY

Made the following visual updates to the explore datasource panel:
- Changed "Dataset" to "Chart Source"
  - Increased font size to 14px
- Removed "More dataset related options" tooltip
- Moved inner collapse icon to the right by 8px
- Removed outer dataset icon when panel is collapsed
- Changed icon to ConsoleSqlOutlined when chart source is a Query
- User sees beginning of SQL query when chart source is a Query

#### When chart source is a Query:
panelUpdateAfter

#### When chart source is not a Query:
panelUpdateAfter2

#### Collapsed panel (no longer has an icon):
panelUpdateAfter3

### TESTING INSTRUCTIONS

- Go to explore by opening any chart
- Observe the visual updates

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx [hunk 1]
```diff
@@ -244,9 +244,23 @@ class DatasourceControl extends React.PureComponent {
...
-          {/* Add a tooltip only for long dataset names */}
-          {!isMissingDatasource && datasource.name.length > 25 ? (
+          {isValidDatasourceType ? (
+            <Icons.ConsoleSqlOutlined className="datasource-svg" />
+          ) : (
+            <Icons.DatasetPhysical className="datasource-svg" />
+          )}
+          {isValidDatasourceType ? (
+            /* Add a tooltip only for long dataset names */
+            !isMissingDatasource && datasource.sql.length > 25 ? (
```
Comment (pkdotson) [inline superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:254]: curious here why datasource.sql.length have to be greater than 25?

Comment (lyndsiWilliams) [reply superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:254]: This was how it was initially set, I just added the query route. I'm also not sure why 25 haha

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx [hunk 2]
```diff
@@ -244,9 +244,23 @@ class DatasourceControl extends React.PureComponent {
...
         <div className="data-container">
-          <Icons.DatasetPhysical className="dataset-svg" />
-          {/* Add a tooltip only for long dataset names */}
-          {!isMissingDatasource && datasource.name.length > 25 ? (
+          {isValidDatasourceType ? (
+            <Icons.ConsoleSqlOutlined className="datasource-svg" />
+          ) : (
+            <Icons.DatasetPhysical className="datasource-svg" />
+          )}
+          {isValidDatasourceType ? (
```
Comment (hughhhh) [inline superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx]: we need to check if the datasource.type is a `DatasourceType.Query` here, if so then show the image in sql as the name

Comment (lyndsiWilliams) [reply superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx]: I changed this to be a strict check for just Query in [`this commit`](https://github.com/apache/superset/pull/20317/commits/6fab57f9ddac90ed6a4aefe028ea1f66905858d0) 😁

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx [hunk 3]
```diff
@@ -244,9 +244,23 @@ class DatasourceControl extends React.PureComponent {
     return (
       <Styles data-test="datasource-control" className="DatasourceControl">
         <div className="data-container">
-          <Icons.DatasetPhysical className="dataset-svg" />
-          {/* Add a tooltip only for long dataset names */}
-          {!isMissingDatasource && datasource.name.length > 25 ? (
+          {datasource.type === DatasourceType.Query ? (
```
Comment (zhaoyongjie) [inline superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:247]: same before

File: superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx (hunk 1 above)
Comment (eschutho) [reply superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx:254]: can we make this value a constant then?

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx [hunk 4]
```diff
@@ -307,11 +307,7 @@ export default function DataSourcePanel({
     return true;
   };
 
-  const isValidDatasourceType =
-    datasource.type === DatasourceType.Dataset ||
-    datasource.type === DatasourceType.SlTable ||
-    datasource.type === DatasourceType.SavedQuery ||
-    datasource.type === DatasourceType.Query;
+  const dataSourceIsQuery = datasource?.type === DatasourceType.Query;
```
Comment (eschutho) [inline superset-frontend/src/explore/components/DatasourcePanel/index.tsx:310]: can we duck type this instead?

**Comment by lyndsiWilliams** [comment]
This got copied into this feature branch: https://github.com/apache/superset/pull/20281

**Pull request closed without merging**